tests/test_csr.py

```diff
@@ -241,3 +246,193 @@ def test_admin_save_existing_csr_should_not_renew(admin_client):
     # CSR and PK should not be regenerated
     assert signing_request.csr == original_csr
     assert signing_request.private_key == original_pk
+
+
+@pytest.mark.django_db
+def test_saving_valid_cert_does_create_cert_instance_via_post(
+    admin_client,
+    temp_private_root,
+):
+    assert Certificate.objects.count() == 0
+
+    csr = SigningRequest.objects.create(
+        common_name="test.example.com",
+        organization_name="Test Organization",
+        state_or_province_name="Test State",
+        country_name="NL",
+        email_address="email@valid.com",
+    )
+
+    private_key = load_pem_x509_private_key(csr.private_key.encode())
+    # Here I did not use the fixture mkcert to generate the certificate
+    # This is because the fixture mkcert did not generate a valid certificate
+    # The public keys of the private key and the certificate did not match
```

Comment:
the `mkcert` allows us to pass in the subject and issuer key - what exactly is not working? And I'd rather add more options to `mkcert` then instead, now this seems to just be duplicating a bunch of code.

IMO we should be able to at least re-use the `root_key` and `root_cert` fixtures for the `issuer` parameters in `mkcert`.

Maybe an optional `csr` parameter needs to be added so you can properly copy over the subject details from the CSR.

Reply:
lol I don't really remember why I couldn't use it before cause I could make it fit pretty easily this time.